OutOfSchool/OutOfSchool.WebApi/Controllers/V1/ParentController.cs

```diff
@@ -60,4 +62,34 @@ public async Task<ActionResult<ParentDTO>> GetProfile()
         var parentDto = await serviceParent.GetByUserId(userId).ConfigureAwait(false);
         return parentDto is not null ? Ok(parentDto) : NoContent();
     }
-}
\ No newline at end of file
+
+    /// <summary>
+    /// Block or unblock Parent entity based on the provided information.
+    /// </summary>
+    /// <param name="parentBlockByAdmin">A DTO containing the necessary information to block or unblock a parent,
+    /// including the ParentId, the desired block status, and a reason.</param>
+    /// <returns>A <see cref="Task{TResult}"/> representing the result of the asynchronous operation.</returns>
+    [HasPermission(Permissions.ParentBlock)]
+    [HttpPost("BlockParent")]
+    [ProducesResponseType(StatusCodes.Status200OK)]
+    [ProducesResponseType(StatusCodes.Status400BadRequest)]
+    [ProducesResponseType(StatusCodes.Status403Forbidden)]
+    [ProducesResponseType(StatusCodes.Status404NotFound)]
+    [ProducesResponseType(StatusCodes.Status500InternalServerError)]
+    public async Task<ActionResult> BlockParent(ParentBlockByAdminDto parentBlockByAdmin)
+    {
+        var result = await serviceParent.BlockParent(parentBlockByAdmin).ConfigureAwait(false);
+
+        if (result.Succeeded)
+        {
+            return Ok();
+        }
+
+        if (result.OperationResult.Errors.Any(x => x.Code == "404"))
```

Comment:
(Code style) Please avoid usage of literals in code. Use constants instead. I'm talking about `"404"`.